URLs.md

```diff
@@ -0,0 +1,105 @@
+
+Following the chain ID may be a [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) account specifier, a [CAIP-19](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md) asset specifier, a transaction identifier, or other unspecified identifiers for on-chain resources and activity.
+
+```
+chain://<network>
+chain://<network>:<address>
+chain://<network>/<asset_namespace>:<asset_reference>/...
+chain://<network>/tx:<tx_identifier>
+chain://<network>/<future_placeholder>
+
+network             = CAIP-2 chain id
+address             = an address string in the common format of the given network
+asset_namespace     = an identifier for a class of related assets, such as erc20 and slip44
+asset_reference     = identifies an asset within its asset_namespace
+tx_identifier       = a transaction identifier in the common format of the given network
+future_placeholder  = future message types
+...                 = addition url parameters for the standard (optional)
+```
+
+Examples:
+
+```
+chain://eip155:1:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03
+chain://eip155:1/erc721:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03/7894
+chain://bip122:000000000019d6689c085ae165831e93/slip44:0
+```
+
+While the protocol allows for the introduction of new on-chain resources, the minimal set implemented by a node or a client must include:
+
+- Ethereum addresses
+- Ethereum transactions
+- Ethereum ERC20 contracts
+- Individual Ethereum ERC-721 tokens
+- Individual Ethereum ERC-1155 tokens
+
+### Farcaster Resources
+
+Farcaster URLs must be capable of identifying any message signed by a user. A sample URL might look like `farcaster://id:8789213729/cast:0xf00b4r` and is of the form:
+
+```
+farcaster://<user>
+farcaster://<user>/<message_type>:<message_id>
+
+user                = id:<ID>
+ID                  = uint256
+message_type        = "cast" | "reaction" | "verified_address" | "follow" | <future_placeholder>
+message_id          = unique identifier for the message
+future_placeholder  = future message types
+```
+
+Example:
+
+```
+farcaster://id:8789213729/cast:0xf00b4r
+```
+
+## Versioning
+
+If an existing resource or activity type already represented by Farcaster URLs must be updated for any reason, the same URL structure must continue to be supported alongside its upgraded version. A hyphenated suffix should describe the version upgrade. For example:
+
+```
+farcaster://id:8789213729/cast-v2:0xf00b4r/1652945621
+```
+
+might represent an update to the URL schema Farcaster Casts that adds a unix epoch timestamp to follow `<message_id>` in the original schema.
+
+Any namespace identifier in the URL schema may be suffixed with a version update, as can the protocol identifier itself if necessary (e.g. `farcaster-v2://`).
+
+## Extending
```

Comment:
I'd remove this section in favor of a more generic "how to extend the farcaster protocol and get involved" section later